src/Arcus.Security.Secrets.Core/Interfaces/ISecretProvider.cs

```diff
@@ -17,6 +18,17 @@ public interface ISecretProvider
         /// <exception cref="ArgumentException">The name must not be empty</exception>
         /// <exception cref="ArgumentNullException">The name must not be null</exception>
         /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        [Obsolete("Use the " + nameof(GetSecret) + " method instead.")]
```

Comment:
Actually, I'm wondering - Do we actually need to deprecate this one?

The only reason why I would remove it maybe is that we are introducing `SecretProvider.GetSecret` but that's fair.

What do you think?

Reply:
For easier access maybe to only get the secret value.
Maybe rename it from `.Get()` to `.GetSecretValue()` then?

Reply:
That's actually a breaking change, let's keep this one deprecated but add these new endpoints in this PR:
- `GetSecret`
- `GetRawSecret` which is new implementation of Get

Reply:
Yes, I know; but the change you requested (adding a `.GetSecret` method to the interface) is also a breaking change to any clients?

Reply:
You mean because it's a new method on an interface or?

Reply:
If so, I think it's ok to consider this a change that is ok to make imo

Reply:
So we could add an extra method to the interface but could not rename the existing one?

Reply:
So we could add an extra method to the interface but could not rename the existing one?

Reply:
So we could add an extra method to the interface but could not rename the existing one?

Reply:
Yes as adding will only break implementing classes. Rename will break implementing and consuming

Reply:
Ah yes OK I see. Thanks for clearing that up.

Reply:
> * `GetSecret`
> * `GetRawSecret` which is new implementation of Get

Ok, I go with your previous example!